sdk/src/Services/S3/Custom/Transfer/Internal/MultipartDownloadManager.cs

```diff
@@ -353,6 +361,9 @@ private async Task<DownloadDiscoveryResult> DiscoverUsingPartStrategyAsync(Cance
             // SEP Part GET Step 2: "send the request and wait for the response in a non-blocking fashion"
             var firstPartResponse = await _s3Client.GetObjectAsync(firstPartRequest, cancellationToken).ConfigureAwait(false);
             
+            if (firstPartResponse == null)
```

Comment:
there was some null pointer exception discovered here in a unit test so added this check